pkg/resourceinterpreter/defaultinterpreter/aggregatestatus.go

```diff
@@ -0,0 +1,209 @@
+package defaultinterpreter
+
+import (
+	"encoding/json"
+	"reflect"
+
+	appsv1 "k8s.io/api/apps/v1"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	extensionsv1beta1 "k8s.io/api/extensions/v1beta1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/klog/v2"
+
+	workv1alpha2 "github.com/karmada-io/karmada/pkg/apis/work/v1alpha2"
+	"github.com/karmada-io/karmada/pkg/util"
+	"github.com/karmada-io/karmada/pkg/util/helper"
+)
+
+type aggregateStatusInterpreter func(
+	object *unstructured.Unstructured, aggregatedStatusItems []workv1alpha2.AggregatedStatusItem, clusters []workv1alpha2.TargetCluster,
+) (*unstructured.Unstructured, error)
+
+func getAllDefaultAggregateStatusInterpreter() map[schema.GroupVersionKind]aggregateStatusInterpreter {
+	s := make(map[schema.GroupVersionKind]aggregateStatusInterpreter)
+	s[appsv1.SchemeGroupVersion.WithKind(util.DeploymentKind)] = aggregateDeploymentStatus
+	s[corev1.SchemeGroupVersion.WithKind(util.ServiceKind)] = aggregateServiceStatus
+	s[extensionsv1beta1.SchemeGroupVersion.WithKind(util.IngressKind)] = aggregateIngressStatus
+	s[batchv1.SchemeGroupVersion.WithKind(util.JobKind)] = aggregateJobStatus
+	return s
+}
+
+func toUnstructured(obj interface{}) (*unstructured.Unstructured, error) {
+	uncastObj, err := runtime.DefaultUnstructuredConverter.ToUnstructured(obj)
+	if err != nil {
+		return nil, err
+	}
+	return &unstructured.Unstructured{Object: uncastObj}, nil
+}
```

Comment:
This is a more common method, I think we can move it to `pkg/util/helper/unstructured.go` file.

Reply:
OK, I plan to clean up the rest of the code that converts Unstructured objects in the next PR.

Reply:
Thanks, I have the same idea.